### PR TITLE
Ukbb changes

### PR DESCRIPTION
**Pull request by avihu111**
- descriptor to caching
- keep keys in Collate
- some other minor changes that we might want to revert or adapt.

### CODE REVIEW

File: fuse/dl/lightning/pl_funcs.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def epoch_end_compute_and_log_losses(pl: pl.LightningModule, mode: str, batch_lo
             else:
                 losses.append(elem[key])
         loss = mean(losses)
-        pl.log(f"{mode}.losses.{key}", loss, on_epoch=True)
+        pl.log(f"{mode}/losses.{key}", loss, on_epoch=True)
```
Comment (avihu111) [inline fuse/dl/lightning/pl_funcs.py:160]: might break compatibility..

File: fuse/data/datasets/caching/samples_cacher.py [hunk 2]
```diff
@@ -96,6 +103,8 @@ def __init__(
             self.delete_cache()
 
         self._audit_kwargs = audit_kwargs
+        if 'ignore_nan_inequality' not in self._audit_kwargs:
```
Comment (mosheraboh) [inline fuse/data/datasets/caching/samples_cacher.py:106]: already handled inside audit. please remove.

File: fuse/data/datasets/caching/samples_cacher.py [hunk 3]
```diff
@@ -338,7 +348,7 @@ def _cache(self, orig_sample_id: Any):
         return output_info
 
 
-def _get_available_write_location(cache_dirs: List[str], max_allowed_used_space=0.95):
+def _get_available_write_location(cache_dirs: List[str], max_allowed_used_space=None):
```
Comment (mosheraboh) [inline fuse/data/datasets/caching/samples_cacher.py:351]: revert

File: fuse/data/datasets/dataset_default.py [hunk 4]
```diff
@@ -43,6 +38,7 @@ def __init__(
         dynamic_pipeline: Optional[PipelineDefault] = None,
         cacher: Optional[SamplesCacher] = None,
         allow_uncached_sample_morphing: bool = False,
+        always_return_dict = True,
```
Comment (mosheraboh) [inline fuse/data/datasets/dataset_default.py:41]: @michalozeryflato , is it your change? Can we drop it?

File: fuse/data/utils/collates.py [hunk 5]
```diff
@@ -40,6 +40,7 @@ class CollateDefault(CollateToBatchList):
     def __init__(
         self,
         skip_keys: Sequence[str] = tuple(),
+        keep_keys: Sequence[str] = tuple(),
```
Comment (mosheraboh) [inline fuse/data/utils/collates.py:43]: Document it with the fact that you skip the ones that do not exist.

File: fuse/dl/lightning/pl_funcs.py (hunk 1 above)
Comment (mosheraboh) [reply fuse/dl/lightning/pl_funcs.py:160]: Lets add flag for backward Compatability

File: fuse/utils/data/collate.py [hunk 6]
```diff
@@ -99,7 +94,7 @@ def _collect_values_to_list(self, samples: List[str], key: str) -> Tuple[List, b
                 if self._raise_error_key_missing:
                     raise Exception(f"Error: key {key} does not exist in sample {index}: {sample}")
                 else:
-                    value = None
+                    continue
```
Comment (mosheraboh) [inline fuse/utils/data/collate.py:97]: Seems like it can cause an issue. Lets revert?

File: fuseimg/data/ops/aug/geometry.py [hunk 7]
```diff
@@ -49,6 +49,8 @@ def __call__(
         :param interpolation: used PIL interpolation method see TTF.affine for details.
         :return: the augmented image
         """
+        if key not in sample_dict:
```
Comment (mosheraboh) [inline fuseimg/data/ops/aug/geometry.py:52]: Let's wrap with OpIfExist which applies an op if some key exists. WDYT?
You will need to implement it :-P

**Review by mosheraboh — CHANGES_REQUESTED** [review]
Looks good!
I've added some suggestions and comments inline.

File: fuseimg/data/ops/aug/geometry.py (hunk 7 above)
Comment (avihu111) [reply fuseimg/data/ops/aug/geometry.py:52]: I'll revert for now.
Not using it anyway.

File: fuse/data/datasets/dataset_default.py (hunk 4 above)
Comment (avihu111) [reply fuse/data/datasets/dataset_default.py]: @itaijj can you explain why this is needed?

**Review by mosheraboh — APPROVED** [review]
Thanks a lot. Looks great.

**Review by mosheraboh — APPROVED** [review]

**Pull request merged**